services/gateway/shared/utils.js

```diff
@@ -93,9 +93,16 @@ const requireAllJson = apiName => {
 
 const getSwaggerDescription = params => `${params.description}\n RPC => ${params.rpcMethod}`;
 
+const isValidResponse = res => {
+	if (Array.isArray(res.data) && res.data.length) return true;
```

Comment:
An empty array within the `data` is also a valid response. If we want not to cache the empty responses, I'd recommend updating the method name to `isValidNonEmptyResponse` or something similar.

```suggestion
	if (Array.isArray(res.data)) return true;
```

Reply:
Updated the method name to `isValidNonEmptyResponse`.